torch/csrc/jit/runtime/symbolic_script.cpp

```diff
@@ -1014,63 +1014,26 @@ const std::vector<std::string> functions = {
             return output, backward
 
         # disable the layernorm AD temporarily because of bug in https://github.com/pytorch/pytorch/issues/19769
-        def layer_norm_disabled(input : Tensor,
+        def layer_norm(input : Tensor,
                        normalized_shape : List[int],
                        weight : Optional[Tensor],
                        bias : Optional[Tensor],
                        eps : float,
                        cudnn_enable : bool):
 
-            input_ndim = input.dim()
-            normalized_ndim = len(normalized_shape)
-            n = 1
-            for i in range(input_ndim - normalized_ndim):
-                n *= input.size(i)
-
-            input_reshape = input.contiguous().view(1, n, -1)
-
-            bn_out, save1, save2, reserve, impl_idx = torch._batch_norm_impl_index(
-                input_reshape, None, None, None, None, True,
-                0.0, eps, cudnn_enable)
-
-            bn_out = bn_out.view(input.size())
-            if weight is not None and bias is not None:
-                output = bias.addcmul(bn_out, weight, value=1)
-            elif weight is not None:
-                output = bn_out.mul(weight)
-            elif bias is not None:
-                output = bn_out.add(bias)
-            else:
-                output = bn_out
+            output, mean, rstd = torch.native_layer_norm(input, normalized_shape, weight, bias, eps)
 
             def backward(grad_output):
```

Comment:
Backward is a little bit scary here, let's add a test for the non-jit code path with combination of gradient for (input, weight, bias).

Reply:
You want to add a test in `test_jit_cuda_fuser.py`

Reply:
We are only testing symbolic_script, so maybe `test_jit.py` instead.

Reply:
oops, I realized that I was negligent earlier and didn't get it that we don't have a simple test in `test_jit_cuda_fuser.py` for added parser rule. :(

Let's at least throw a simple one with `native_layer_norm`.
Slightly awkward on modular testing for backward nodes, as they are not exposed in python level :(